### DISCUSSION TITLE
Django celery apply_async running multiple times

### DISCUSSION THREAD

**Discussion by anuj9196** [celery/celery#7482] [Q&A]
Using **Django 2.2** and **Celery 4.4.7**.

Using **AWS ECS** to deploy the application and celery and using **SQS** as message broker.

In the ECS task definition, using the following command to start the celery task
```
["celery","worker","-A","app","-l","info","--concurrency","4"]
```

There are 8 tasks running for the celery service and 4 tasks for the application

[![enter image description here][1]][1]

When using `apply_async` to delay a task 
```python
send_webhook_task.apply_async(
    kwargs={
        'tracking_id': tracking_data.id,
        'hook_url': hook.url
    },
    countdown=60
)
```
The above task is executed after 60 seconds, but it executes for 3-4 times and the same data is sent multiple times.

Same when using `delay()` is executed only once
```python
send_webhook_task.delay(
    tracking_id=tracking_data.id,
    hook_url=hook.url
)
```

**Why `apply_sync` is executing multiple times with the above setup?**

Celery broker settings
```python
CELERY_BROKER_TRANSPORT_OPTIONS = {
    'visibility_timeout': 1500,
    'polling_interval': 60
}
```

  [1]: https://i.stack.imgur.com/duInT.png

**Comment by Tiwari1812** [comment]
@anuj9196 did you get any answer as i am also stuck in a similar situation

**Comment by tribumirkov** [comment]
me too!

**Comment by auvipy** [comment]
we stopped celery v4.x long ago.

**Reply by tribumirkov** [reply]
hey Asif, thank you for you reply! I'm experiencing the same issue on 5.2.7, should I report a separate issue?

**Reply by auvipy** [reply]
no you can describe here

**Reply by tribumirkov** [reply]
something like this, Asif:

from celery import Celery, chain
from celery.app import shared_task


app = Celery('cronjobs')
app.conf.broker_url = "redis://localhost:6379"
app.conf.result_backend = "redis://localhost:6379"


@shared_task(name='task_1')
def task_1():
    """
    Task 1
    """
    print('Task 1: doing something...')


@shared_task(name='task_2')
def task_2():
    """
    Task 2
    """
    print('Task 2: doing something else...')


@app.on_after_configure.connect
def setup_periodic_tasks(sender, **kwargs):
    """
    Setup periodic tasks
    """
    sender.add_periodic_task(
        10,
        chain(
            task_1.s(),
            task_2.si()
        ).apply()
    )

and I run "celery -A app beat --loglevel=info" and "celery -A app worker --loglevel=info"
